apps/gateway/src/chat/chat.ts

```
@@ -746,10 +746,25 @@
  * Transforms streaming chunk to OpenAI format for non-OpenAI providers
  */
+// Helper function to calculate prompt tokens when missing or 0
+function calculatePromptTokensFromMessages(messages: any[]): number {
+	try {
+		const chatMessages: ChatMessage[] = messages.map((m: any) => ({
+			role: m.role,
+			content: typeof m.content === "string" ? m.content : JSON.stringify(m.content),
+			name: m.name,
+		}));
+		return encodeChat(chatMessages, DEFAULT_TOKENIZER_MODEL).length;
+	} catch (error) {
```

ESLint

[error] 'error' is defined but never used. Allowed unused caught errors must match /^_/u.

(no-unused-vars)